examples/java/pom.xml

```diff
@@ -367,10 +367,6 @@
 
     <plugins>
       <plugin>
-        <artifactId>maven-compiler-plugin</artifactId>
```

Comment:
please update Commit description to explain why this is correct. I believe you're saying that compiler-plugin, jar-plugin, and checkstyle are already on by default for all modules so need not be explicitly enabled?

Confirm by analyzing the difference in `mvn help:effective-pom`.